packages/docusaurus-init/templates/classic-typescript/tsconfig.json

```diff
@@ -0,0 +1,10 @@
+{
+  // This file is not used in compilation. It is here just for a nice editor experience.
+  "compilerOptions": {
+    "jsx": "react",
+    "esModuleInterop": true
+  },
+  "extends": "@tsconfig/docusaurus/tsconfig.json",
+  "include": ["src/"],
+  "exclude": ["templates/"]
+}
```

Comment:
should we move some of those options in `@tsconfig/docusaurus` ?

Reply:
`"exclude": ["templates/"]` is in the wrong place. Thanks for noticing.
IDK what _should_ be in `@tsconfig/docusaurus`, but if you do add these options in the base config then let's remove them here

Reply:
> IDK what should be in @tsconfig/docusaurus, but if you do add these options in the base config then let's remove them here

The question is: are those options here mandatory? Why did you add them exactly? If those are required then we definitively need to move them in the base config

Reply:
I added these:

```ts
"compilerOptions": {
    "jsx": "react",
    "esModuleInterop": true,
  },
```

 to clear all build errors. Considering that `website/tsconfig.json` also has these two fields, I would say they are mandatory and can be added to the base config.

But `noEmit: true` was only added because I saw you added the `typecheck` script which resolves to `tsc`, so this is optional.

Reply:
The base config already contains all those so I suggest we could just remove everything and just extend the base config?

https://github.com/tsconfig/bases/blob/main/bases/docusaurus.json

BTW for dogfooding it would also be great to make our website use this base config, and what about converting our homepage to TS too? (can be done in another PR)

Reply:
Definitely, I can help with this, maybe the other day